core-plugins/src/main/java/io/cdap/plugin/batch/action/FileDeleteAction.java

```diff
@@ -126,13 +131,13 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     @Description("Indicates if the pipeline should continue if the delete fails")
     private boolean continueOnError;
 
-    public void validate() {
+    public void validate(FailureCollector collector) {
       if (!containsMacro("fileRegex") && fileRegex != null) {
         try {
           Pattern.compile(fileRegex);
         } catch (Exception e) {
-          throw new IllegalArgumentException(String.format("File regex %s is invalid: %s",
-                                                           fileRegex, e.getMessage()), e);
+          collector.addFailure(String.format("File regex %s is invalid: %s", fileRegex, e.getMessage()),
+                  "Provide valid regex.").withConfigProperty("fileRegex");
```

Comment:
create constants for these, similar comment elsewhere

Reply:
Done